[CI] ESNextDiskBBQVectorsFormatTests testSlicesDenseWithFilter failing

**Build Scans:**
- [elasticsearch-periodic #12493 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/kthdbomwafyjq)
- [elasticsearch-periodic-platform-support #12466 / rhel-10_platform-support-unix](https://gradle-enterprise.elastic.co/s/peim3vandw7me)
- [elasticsearch-pull-request #162062 / part-1](https://gradle-enterprise.elastic.co/s/i6kof7nigjzxc)
- [elasticsearch-pull-request #162062 / part-1](https://gradle-enterprise.elastic.co/s/3buepfauzcx3s)
- [elasticsearch-pull-request #162038 / part-1](https://gradle-enterprise.elastic.co/s/qtlkwscr23g26)
- [elasticsearch-pull-request #162038 / part-1](https://gradle-enterprise.elastic.co/s/64uewv7pdjpsq)
- [elasticsearch-intake #43122 / part1](https://gradle-enterprise.elastic.co/s/ixjy34j7yl5kk)
- [elasticsearch-pull-request #162006 / part-1](https://gradle-enterprise.elastic.co/s/hv2lkx7fn5msk)
- [elasticsearch-pull-request #161878 / part-1](https://gradle-enterprise.elastic.co/s/2bofjo4klqckw)
- [elasticsearch-pull-request #161878 / part-1](https://gradle-enterprise.elastic.co/s/nvfitafrta3io)

**Reproduction Line:**
```
null
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.codec.vectors.diskbbq.next.ESNextDiskBBQVectorsFormatTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testSlicesDenseWithFilter'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: assignments and overspillAssignments must have the same length
```

**Issue Reasons:**
- [main] 1 consecutive failures in step concurrent-search-tests
- [main] 1 consecutive failures in step rhel-10_platform-support-unix
- [main] 1 consecutive failures in pipeline elasticsearch-periodic
- [main] 11 failures in test testSlicesDenseWithFilter (3.3% fail rate in 332 executions)
- [main] 7 failures in step part-1 (3.1% fail rate in 225 executions)
- [main] 2 failures in step part1 (2.7% fail rate in 75 executions)
- [main] 4 failures in pipeline elasticsearch-pull-request (1.8% fail rate in 222 executions)
- [main] 2 failures in pipeline elasticsearch-intake (2.7% fail rate in 75 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -793,9 +793,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.cache.SharedBlobCacheWarmingServiceTests
   method: testShardRecoveryWarmingPropagatesTimestamp
   issue: https://github.com/elastic/elasticsearch/issues/153223
-- class: org.elasticsearch.index.codec.vectors.diskbbq.next.ESNextDiskBBQVectorsFormatTests
-  method: testSlicesDense
-  issue: https://github.com/elastic/elasticsearch/issues/153272
 - class: org.elasticsearch.xpack.stateless.reshard.StatelessReshardIT
   method: testTermVectorsApiRealtimeGet
   issue: https://github.com/elastic/elasticsearch/issues/153282
@@ -847,9 +844,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.parquet.ParquetTestingIT
   method: testParquetFile {column_chunk_key_value_metadata}
   issue: https://github.com/elastic/elasticsearch/issues/153361
-- class: org.elasticsearch.index.codec.vectors.diskbbq.next.ESNextDiskBBQVectorsFormatTests
-  method: testSlicesDenseWithFilter
-  issue: https://github.com/elastic/elasticsearch/issues/153362
 - class: org.elasticsearch.xpack.esql.qa.parquet.ParquetTestingIT
   method: testParquetFile {binary}
   issue: https://github.com/elastic/elasticsearch/issues/153364

--- a/server/src/main/java/org/elasticsearch/index/codec/vectors/cluster/KMeansWithOverspill.java
+++ b/server/src/main/java/org/elasticsearch/index/codec/vectors/cluster/KMeansWithOverspill.java
@@ -117,6 +117,7 @@ public record KMeansWithOverspill<V>(KMeansResult<V> result, OverspillAssignment
             if (index < 0) {
                 index = -index - 2; // go back one to the offset < ordinal, as that's the assignment containing this ordinal
             }
+            // if this ordinal is out of range for the indexed assignment, that's ok, it'll just return an empty iterator
             var iterator = assignments[index].getAssignmentsFor(ordinal - assignmentOffsets[index]);
             return new MappingIntIterator(iterator, centroidOffsets[index]);
         }

